### PR TITLE
Fix lakectl local diff

### PR DESCRIPTION
**Pull request by idanovo**
Closes #(Insert issue number closed by this PR)

### Background

Share context and relevant information for the PR: offline discussions, considerations, design decisions etc.

### Bug Fix

If this PR is a bug fix, please let us know about:

1. Problem - The reason for opening the bug
2. Root cause - Discovered root cause after investigation
3. Solution - How the bug was fixed
      
### New Feature

If this PR introduces a new feature, describe it here.

### Testing Details

How were the changes tested?

### Breaking Change?

Does this change break any existing functionality? (API, CLI, Clients)

## Additional info

Logs, outputs, screenshots of changes if applicable (CLI / GUI changes)

### Contact Details

How can we get in touch with you if we need more info? (ex. email@example.com)

### CODE REVIEW

File: pkg/local/diff.go [hunk 1]
```diff
@@ -194,14 +197,29 @@ func DiffLocalWithHead(left <-chan apigen.ObjectStats, rightPath string) (Change
...
+	if err != nil {
+		return nil, err
+	}
+	err = reader.Walk(context.Background(), &uri, block.WalkOptions{}, func(e block.ObjectStoreEntry) error {
+		info, err := os.Stat(e.FullKey)
 		if err != nil {
 			return err
 		}
-		if info.IsDir() || diffShouldIgnore(info.Name()) {
+		if info.IsDir() || diffShouldIgnore(info.Name()) || strings.HasPrefix(e.RelativeKey, local.CacheDirName) {
```
Comment (N-o-Z) [inline pkg/local/diff.go:218]: We don't actually use the cache so we can just pass an empty path and remove this

File: pkg/local/diff.go [hunk 2]
```diff
@@ -194,14 +197,29 @@ func DiffLocalWithHead(left <-chan apigen.ObjectStats, rightPath string) (Change
...
 			return nil
 		}
-		localPath := strings.TrimPrefix(path, rightPath)
+		filePath := e.RelativeKey
+		localPath := strings.TrimPrefix(filePath, rightPath)
```
Comment (N-o-Z) [inline pkg/local/diff.go:222]: Why do we need this? Also if we do, do we need the next line then?

File: pkg/block/local/walker.go [hunk 3]
```diff
@@ -19,7 +19,7 @@ import (
 	"github.com/treeverse/lakefs/pkg/block/params"
 )
 
-const cacheDirName = "_lakefs_cache"
+const CacheDirName = "_lakefs_cache"
```
Comment (N-o-Z) [inline pkg/block/local/walker.go:22]: Can revert this as we don't really use the cache

File: pkg/local/diff.go [hunk 4]
```diff
@@ -194,14 +197,29 @@ func DiffLocalWithHead(left <-chan apigen.ObjectStats, rightPath string) (Change
 		currentRemoteFile apigen.ObjectStats
 		hasMore           bool
 	)
-	err := filepath.Walk(rightPath, func(path string, info fs.FileInfo, err error) error {
+	absPath, err := filepath.Abs(rightPath)
+	if err != nil {
+		return nil, err
+	}
+	uri := url.URL{Scheme: "local", Path: absPath}
+	adapter, err := local.NewAdapter(absPath, local.WithRemoveEmptyDir(false), local.WithAllowedExternalPrefixes([]string{"/"}))
```
Comment (N-o-Z) [inline pkg/local/diff.go:205]:
```suggestion
	adapter, err := local.NewAdapter("", local.WithRemoveEmptyDir(false), local.WithAllowedExternalPrefixes([]string{absPath}))
```

File: esti/lakectl_local_test.go [hunk 5]
```diff
@@ -437,12 +437,15 @@ func TestLakectlLocal_commit(t *testing.T) {
...
-
 			RunCmdAndVerifyContainsText(t, Lakectl()+" local status "+dataDir, false, "No diff found", vars)
 
 			// Modify local folder - add and remove files
+			os.MkdirAll(filepath.Join(dataDir, "subdir"), os.ModePerm)
+			os.MkdirAll(filepath.Join(dataDir, "subdir-a"), os.ModePerm)
+			fd, err = os.Create(filepath.Join(dataDir, "subdir", "test.txt"))
+			require.NoError(t, err)
+			fd, err = os.Create(filepath.Join(dataDir, "subdir-a", "test.txt"))
+			require.NoError(t, err)
```
Comment (N-o-Z) [inline esti/lakectl_local_test.go:448]: I think this is missing verification for this test case (on the status)

File: esti/lakectl_test.go [hunk 6]
```diff
@@ -557,7 +557,7 @@ func TestLakectlFsUpload(t *testing.T) {
 	})
 	t.Run("single_file_with_recursive", func(t *testing.T) {
 		vars["FILE_PATH"] = "data/ro/ro_1k.0"
-		RunCmdAndVerifyFailure(t, Lakectl()+" fs upload --recursive -s files/ro_1k lakefs://"+repoName+"/"+mainBranch+"/"+vars["FILE_PATH"], false, "\ndiff 'local://files/ro_1k/' <--> 'lakefs://${REPO}/${BRANCH}/${FILE_PATH}'...\nlstat files/ro_1k/: not a directory\nError executing command.\n", vars)
```
Comment (N-o-Z) [inline esti/lakectl_test.go:560]: How/Why was this changed?

**Review by N-o-Z — CHANGES_REQUESTED** [review]
Great job!
Some comments on the code and tests.
**One important thing I think we have to do before merging:**
Lets check (manually) that we did not break compatibility with windows...

File: esti/lakectl_test.go (hunk 6 above)
Comment (N-o-Z) [reply esti/lakectl_test.go:560]: Let's not do that, let's not use caching (see other comments)

File: esti/lakectl_local_test.go (hunk 5 above)
Comment (idanovo) [reply esti/lakectl_local_test.go:448]: At the end of the test, we verify that lakectl status returned no diff.
In the original bug, we've got changes after committing

Comment (N-o-Z) [reply esti/lakectl_local_test.go:448]: Understood, but we still need to fix the verify status output

**Review by N-o-Z — APPROVED** [review]
Awesome!
BTW this probably fixed a bug also with local storage import on Windows 🎉
